docs/plugins/create/howto.md

```diff
@@ -133,11 +133,45 @@ function PLUGIN:EnvKeys(ctx)
 end
 ```
 
+## PreUse
+
+When the user uses `vfox use`, the plugin's `PreUse` function is called. The purpose of this function is to return the
+version information entered by the user. If the `PreUse` function returns version information, `vfox` will use this new
+version.
+
+```lua
+function PLUGIN:PreUse(ctx)
+    local runtimeVersion = ctx.runtimeVersion
+    --- user input version
+    local version = ctx.version
+    --- user current used version
+    local previousVersion = ctx.previousVersion
+
+    --- installed sdks
+    local sdkInfo = ctx.installedSdks['version']
```

Comment:
I don't think we need to pass information about all installed SDKs, just the current version and the previous version is enough.  

eg: 
```lua
local curentSdkInfo = ctx.currentSdk
local path = curentSdkInfo.path
local name = curentSdkInfo.name
local version = curentSdkInfo.version

local previousSdkInfo = ctx.previousSdk
local path = previousSdkInfo.path
local name = previousSdkInfo.name
local version = previousSdkInfo.version
```

In addition, `use` command is a frequently used command. If we get all installed versions every time, if there are many, it will slightly affect the experience.

Thought?

Reply:
the reason I want this hooks is that I want use a shorthand version instead of the full version.

`vfox use nodejs@20`

so we need let the plugins knows what sdks we currently have.

and the thing you say can be `PostUse`, here we have the previous version and current version

Reply:
Oops, I forgot about this scene. ;) 

Let's keep it that way.

Reply:
I determined whether `PreUse` hook exists and then obtained installed versions, this can reduce some effect.